lib/matplotlib/cbook.py

```diff
@@ -361,9 +361,11 @@ class _BoundMethodProxy(object):
     Minor bugfixes by Michael Droettboom
     '''
     def __init__(self, cb):
+        self._hash = hash(cb)
+        self._callbacks = []
```

Comment:
Well, they do get uniquely identified, the problem lies in that they have too many unique identifiers, they should have just one, each (signal, callback) pair should have a single unique id.  This does not happen.

As a knock on effect, if you were to trigger the signal in the test cases, the callback would get called twice, when it should only get called the once.